docs/pages/pmd/devdocs/major_contributions/adding_new_cpd_language.md

```diff
@@ -8,16 +8,62 @@ permalink: pmd_devdocs_major_adding_new_cpd_language.html
 author: Romain PELISSE <belaran@gmail.com>
 ---
 
-If you wish CPD to parse a unsupported language, you can easily develop a new parser for CPD. All you need to is implements the following interface:
+First of all, thanks for the contribution!     
+    
+Happily for you, to add CPD support for a new language is now easier than ever!     
+> **Pro Tip**: If you wish to add a new language, there are more than 50 languages you could easily add with just an [Antlr grammar](https://github.com/antlr/grammars-v4).    
 
-*   net.sourceforge.pmd.cpd.Language
-*   net.sourceforge.pmd.cpd.Tokenizer
 
-Do not forget to the follow the proper naming convention, as the CPD parser factory use this convention:
+ All you need to do is follow this few steps:
+     
+1. Create a new module for your language, you can take [GO as an example](https://github.com/pmd/pmd/tree/master/pmd-go)    
+2. Create a Tokenizer
+	
+- For Antlr grammars you can take the grammar from [here](https://github.com/antlr/grammars-v4)  and extend [AntlrTokenizer](https://github.com/pmd/pmd/blob/master/pmd-core/src/main/java/net/sourceforge/pmd/cpd/AntlrTokenizer.java)  taking Go as an example
+  
+	
+```java 
+  public class GoTokenizer extends AntlrTokenizer {    
+    
+      @Override protected AntlrTokenManager getLexerForSource(SourceCode sourceCode) {   
+          CharStream charStream = AntlrTokenizer.getCharStreamFromSourceCode(sourceCode);   
+          return new AntlrTokenManager(new GolangLexer(charStream), sourceCode.getFileName());   
+      }
+  }
+```
 
-*   Language Name + "Language"
-*   Tokenizer Name + "Tokenizer"
+	
+- For any other scenario you can use [AnyTokenizer](https://github.com/pmd/pmd/blob/master/pmd-core/src/main/java/net/sourceforge/pmd/cpd/AnyTokenizer.java)
```

Comment:
There is also the JavaCC scenario. Here you'd subclass JavaCCTokenizer. And example would be PythonTokenizer.

Reply:
Thanks! I'll add that scenario!